src/libaktualizr/bootloader/bootloader.cc

```diff
@@ -37,10 +42,30 @@ inline void CopyFromConfig(RollbackMode& dest, const std::string& option_name, c
 
 void BootloaderConfig::updateFromPropertyTree(const boost::property_tree::ptree& pt) {
   CopyFromConfig(rollback_mode, "rollback_mode", pt);
+  CopyFromConfig(reboot_sentinel_dir, "reboot_sentinel_dir", pt);
+  CopyFromConfig(reboot_sentinel_name, "reboot_sentinel_name", pt);
 }
 
 void BootloaderConfig::writeToStream(std::ostream& out_stream) const {
   writeOption(out_stream, rollback_mode, "rollback_mode");
+  writeOption(out_stream, reboot_sentinel_dir, "reboot_sentinel_dir");
+  writeOption(out_stream, reboot_sentinel_name, "reboot_sentinel_name");
+}
+
+Bootloader::Bootloader(const BootloaderConfig& config, INvStorage& storage) : config_(config), storage_(storage) {
+  reboot_sentinel_ = config_.reboot_sentinel_dir / config_.reboot_sentinel_name;
+
+  if (mkdir(config_.reboot_sentinel_dir.c_str(), S_IRWXU) == -1) {
+    struct stat st {};
+    stat(config_.reboot_sentinel_dir.c_str(), &st);
+    if (((st.st_mode & S_IFDIR) == 0) || (st.st_mode & (S_IRGRP | S_IROTH | S_IWGRP | S_IWOTH)) != 0) {
+      LOG_WARNING << "Could not create " << config_.reboot_sentinel_dir
+                  << " securely, reboot detection support disabled";
+      reboot_detect_supported_ = false;
+      return;
+    }
+  }
+  reboot_detect_supported_ = true;
```

Comment:
Why bother with the second check here? If you can't create the directory for any reason, then there's a problem. If the first test succeeds but the second fails, you are in a strange state that probably won't work but `reboot_detect_supported_` is set to true.

Reply:
If `mkdir` fails but all the conditions checks out, it means that the directory was already there (I hope, you can double check).

The point is to avoid race conditions (better ask forgiveness...)

I'll add a comment to clarify.

Reply:
Perfect, that makes sense.